app/zorgned_service.py

```diff
@@ -0,0 +1,118 @@
+
+    beschikking = dpath_util.get(aanvraag_source, "beschikking", default=None)
+
+    # TODO: Is this a correct assumption?
+    product = dpath_util.get(beschikking, "beschikteProducten/0", default=None)
+
+    if not product:
+        logging.info("Product not found %s" % json.dumps(aanvraag_source, indent=4))
+        return
+
+    toegewezen_product = dpath_util.get(product, "toegewezenProduct", default=None)
+    toewijzing = dpath_util.get(toegewezen_product, "toewijzingen", default=[])
+
+    if toewijzing:
+        toewijzing = toewijzing.pop()
+
+    levering = dpath_util.get(toewijzing, "leveringen", default=[])
+
+    if levering:
+        levering = levering.pop()
+
+    item_type_code = dpath_util.get(product, "product/productsoortCode")
+    if item_type_code:
+        item_type_code = item_type_code.upper()
+
+    delivery_type = dpath_util.get(toegewezen_product, "leveringsvorm", default=None)
+    if delivery_type:
+        delivery_type = delivery_type.upper()
+
+    aanvraag = {
+        # Beschikking
+        "dateDecision": dpath_util.get(beschikking, "datumAfgifte", default=None),
+        # Product
+        "title": dpath_util.get(product, "product/omschrijving"),
+        "itemTypeCode": item_type_code,
+        # Toegewezen product
+        "dateStart": dpath_util.get(
+            toegewezen_product, "datumIngangGeldigheid", default=None
+        ),
+        "dateEnd": dpath_util.get(
+            toegewezen_product, "datumEindeGeldigheid", default=None
+        ),
+        "isActual": dpath_util.get(toegewezen_product, "actueel", default=False),
+        "deliveryType": delivery_type,
+        "supplier": dpath_util.get(
+            toegewezen_product, "leverancier/omschrijving", default=None
+        ),
+        # Levering
+        "serviceOrderDate": dpath_util.get(toewijzing, "datumOpdracht", default=None),
+        "serviceDateStart": dpath_util.get(levering, "begindatum", default=None),
+        "serviceDateEnd": dpath_util.get(levering, "einddatum", default=None),
+    }
+
+    return aanvraag
+
+
+def format_aanvragen(aanvragen_source=[]):
+    aanvragen = []
+
+    for aanvraag_source in aanvragen_source:
+        regeling_id = dpath_util.get(
+            aanvraag_source, "regeling/identificatie", default=None
+        )
+        if regeling_id == REGELING_IDENTIFICATIE:
+            aanvraag_formatted = format_aanvraag(aanvraag_source)
+            if aanvraag_formatted:
+                aanvragen.append(aanvraag_formatted)
+
+    return aanvragen
+
+
+def get_aanvragen(bsn):
+    url = f"{ZORGNED_API_URL}/gemeenten/{ZORGNED_GEMEENTE_CODE}/ingeschrevenpersonen/{bsn}/aanvragen"
+
+    headers = {"Token": ZORGNED_API_TOKEN}
+
+    cert = (SERVER_CLIENT_CERT, SERVER_CLIENT_KEY)
+    res = requests.get(url, timeout=9, headers=headers, cert=cert)
+
+    response_data = res.json()
+
+    logging.debug(json.dumps(response_data, indent=4))
+
+    response_aanvragen = response_data["_embedded"]["aanvraag"]
```

Comment:
Moet hier geen nullcheck gedaan worden?

Reply:
Ik heb er voor gekozen dat niet te doen, indien de Api een response teruggeeft die niet deze 2 keys heeft is er een error en die mag gewoon naar de response propageren d.m.v een 500 internal server error response.